tests/setup.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+set -x
+
+sudo ip tuntap add dev tun10 mode tun user $(whoami)
+sudo ip address add 10.10.10.1/24 dev tun10
+sudo ip link set tun10 up
```

Comment:
This requires `sudo` to be available, as opposed to some other ways of setting this up (eg. running it directly as root). But I guess it's OK to live with it this way; people with really unusual systems might at least use the script as an inspiration.